src/span_processor.ts

```diff
@@ -22,6 +23,9 @@ export class SpanProcessor implements OpenTelemetrySpanProcessor {
   onStart(_span: Span, _parentContext: Context): void {}
 
   onEnd(span: ReadableSpan): void {
+    // Add OpenTelemetry kind enum value as a magic attribute
+    span.attributes["appsignal.kind"] = SpanKind[span.kind]
```

Comment:
Nitpick: I'm wondering if modifying the given OpenTelemetry span would cause problems if there were other span processors -- as in, if it would add our magic attribute for those other span processors as well.

It may be safer to rebuild the attributes object, instead of modifying it in-place in the ReadableSpan object.

```js
const spanAttributes = {
  ...span.attributes,
  "appsignal.kind": SpanKind[span.kind]
}
// then pass this to createOpenTelemetrySpan
```

Reply:
Thanks for this. I prefer your approach; changing it